elide-core/src/main/java/com/yahoo/elide/core/pagination/Pagination.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2015, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.core.pagination;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.ToString;
+
+import javax.ws.rs.core.MultivaluedMap;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Encapsulates the pagination strategy.
+ */
+
+@AllArgsConstructor
+@ToString
+public class Pagination {
+
+    public static final int DEFAULT_PAGE_SIZE = 500;
+    // todo do we ever want to return more than 10000 items? Set Upper Bounds
+
+    private static final Pagination DEFAULT_PAGINATION = new Pagination(0, 0);
+
+    public static final Map<String, String> PAGE_KEYS = new HashMap<>();
+    static {
+        PAGE_KEYS.put("page[size]", "pageSize");
+        PAGE_KEYS.put("page[limit]", "pageSize");
+        PAGE_KEYS.put("page[number]", "page");
+        PAGE_KEYS.put("page[offset]", "page");
+    }
+
+    @Getter
+    private int page;
+
+    @Getter
+    private int pageSize;
+
+    /**
+     * Know if this is the default instance.
+     * @return The default pagination values.
+     */
+    public boolean isDefault() {
+        return this.page == 0 && this.pageSize == 0;
+    }
+
+    /**
+     * Given json-api paging params, generate page and pageSize values from query params.
+     * @param queryParams The page queryParams (ImmuatableMultiValueMap).
+     * @return The new Page object.
+     */
+    public static Pagination parseQueryParams(final MultivaluedMap<String, String> queryParams) {
+        final Map<String, Integer> pageData = new HashMap<>();
+        queryParams.entrySet()
+                .forEach(paramEntry -> {
+                    // looking for page[size], page[limit]
+                    // looking for page[offset], page[number]
+                    final String queryParamKey = paramEntry.getKey();
+                    if (PAGE_KEYS.containsKey(queryParamKey)) {
+                        final String type = PAGE_KEYS.get(queryParamKey);
+                        final String value = paramEntry.getValue().get(0); // is this correct...
+                        try {
+                            pageData.put(type, Integer.parseInt(value, 10));
+                        } catch (ClassCastException e) {
```

Comment:
We should throw a HTTPStatusException with a 400 error.  You could either sublass or use InvalidValueException here.